addons/website_slides/models/slide_embed.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from werkzeug import urls
+
+from odoo import _, fields, models
+
+
+class EmbeddedSlide(models.Model):
+    """ Embedding in third party websites. Track view count, generate statistics. """
+    _name = 'slide.embed'
+    _description = 'Embedded Slides View Counter'
+    _rec_name = 'slide_id'
+
+    slide_id = fields.Many2one('slide.slide', string="Presentation", required=True, index=True)
+    url = fields.Char('Third Party Website URL', required=True)
+    count_views = fields.Integer('# Views', default=1)
+
+    def _embed_increment(self, slide_id, url):
+        """ Increment the view count of the record we have based on the passed url.
+        If the url is empty, which typically happens if the browser does not pass the 'referer'
+        header properly, then we increment the entry inside a common 'Unknown Website' record. """
+
+        if not slide_id or not self.env['slide.slide'].browse(int(slide_id)).exists():
+            raise ValueError(_('Invalid slide_id parameter'))
+
+        url_entry = url
+        if not urls.url_parse(url).netloc:
+            url_entry = 'Unknown Website'
```

Comment:
Maybe translation marks ? That would maybe create several "unknown" entries, but I guess in case of anonymous external website embed I suppose language will always be the default db one ?